File: js/2.js
// By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

var fib = [1, 2];

while (fib[fib.length-1] < 4000000) {
    fib.push(fib[fib.length-1] + fib[fib.length-2]);
}

var total = 0;

for (var i = 0; i < fib.length; i++) {
    if (fib[i] % 2 == 0) {
        total = total + fib[i];
    }
}

console.log(total); // 4613732

File: elixir/2.exs
# By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms. <<<--->>>

defmodule Euler do
  def fib(0), do: 0
  def fib(1), do: 1
  def fib(n), do: fib(n-1) + fib(n-2)
end

result = Stream.iterate(0, fn(n) -> n + 1 end)
      |> Stream.map(fn(n) -> Euler.fib(n) end) 
      |> Stream.filter(fn(x) -> rem(x, 2) == 0 end) 
      |> Stream.take_while(fn(x) -> x < 4000000 end) 
      |> Enum.sum

IO.puts result # 4613732